apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js

```diff
@@ -34,45 +49,136 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
+                        goFurther = true
+                    } else {
+                        const isFeatureFlagEnabled = await featureToggleManager.isFeatureEnabled(
+                            context,
+                            ENABLE_DISCOVER_SERVICE_CONSUMERS,
+                            { organization: organizationId },
+                        )
+
+                        if (isFeatureFlagEnabled) {
+                            goFurther = true
+                        }
+                    }
+
+                    if (goFurther) {
+                        items.push({ organizationId, address, addressKey, unitType, unitName })
+                    }
+                }
+
+                // Filter by acquiring context
+                const organizationsIdsWithAcquiringContext = new Set() //acquiringContexts.map((c) => get(c, ['organization', 'id']))
+                await loadListByChunks({
+                    context,
+                    list: AcquiringIntegrationContext,
+                    chunkSize: 50,
+                    where: {
                         deletedAt: null,
-                    })
+                        organization: { id_in: items.map(({ organizationId }) => organizationId) },
+                        status: CONTEXT_FINISHED_STATUS,
+                    },
+                    chunkProcessor: (chunk) => {
+                        chunk.forEach((row) => {
+                            organizationsIdsWithAcquiringContext.add(get(row, ['organization', 'id']))
+                        })
+                        return []
+                    },
+                })
 
+                items = items.filter((item) => organizationsIdsWithAcquiringContext.has(item.organizationId))
 
-                const getPayload = (accountNumber, resident) => ({
-                    dv,
-                    sender,
-                    resident: { connect: { id: resident.id } },
-                    organization: { connect: { id: get(resident, ['organization', 'id']) } },
-                    accountNumber: accountNumber,
+                // Filter by property
+                const organizationsIdsWithProperties = new Set()
+                await loadListByChunks({
+                    context,
+                    list: Property,
+                    chunkSize: 50,
+                    where: {
+                        deletedAt: null,
+                        organization: { id_in: items.map((item) => item.organizationId) },
+                        OR: items.map((item) => ({
+                            OR: [{ address_i: item.address }, { addressKey: item.addressKey }],
+                        })),
+                    },
+                    chunkProcessor: (chunk) => {
+                        chunk.forEach((row) => {
+                            organizationsIdsWithProperties.add(get(row, ['organization', 'id']))
+                        })
+                        return []
+                    },
```

Comment:
<img width="878" alt="image" src="https://github.com/open-condo-software/condo/assets/50069872/a1255974-f953-4907-b522-e98e29747d12">

Reply:
Fixed

Reply:
This one not solved, since you're filtering items only by organization later

Reply:
also add test